lambda/matching.ml

```diff
@@ -1889,11 +1889,24 @@ let code_force_lazy = get_mod_field "CamlinternalLazy" "force_gen"
    Forward(val_out_of_heap).
 *)
 
+let call_force_lazy_block varg loc =
+  (* The argument is wrapped with [Popaque] to prevent the compiler from making
+     any assumptions on its contents. Alternatively, [ap_inlined] could be set
+     to [Never_inline]. *)
```

Comment:
It would be nice to add a reference here to the PRs #9998 and #10909 that go into the details of why we need Popaque.

(I spend a good share of my review energy asking people to explain decisions more in comments. Maybe I'm more often than most in the position of reading year-old compiler code that I'm unfamiliar with, trying to reverse-engineer detailed justifications for the non-obvious things that are going on?)